src/diffusers/models/unets/unet_2d_blocks.py

```diff
@@ -764,6 +766,10 @@ def __init__(
     ):
         super().__init__()
 
+        out_channels = out_channels or in_channels
+        self.in_channels = in_channels
```

Comment:
do we need this?

Reply:
Yes, we do. Usually, the midblock has the same number of in and out channels. With ControlNet-XS, we need to have a different number of in & out channels, because we're adding extra information from the control model.